modifytags/modifytags.go

```diff
@@ -115,7 +115,11 @@ func (mod *Modification) rewrite(fset *token.FileSet, node ast.Node, start, end
 				continue
 			}
 
-			f.Tag.Value = res
+			if res == "" {
```

Comment:
I think it would be clearer to defer mutating f.Tag until this point. In other words, on line 104 above, just do something like

```go
curTag := ""
if f.Tag != nil {
  curTag = f.Tag.Value
}
```

And then here, do:

```
if res == "" {
  f.Tag = nil
} else {
  if f.Tag == nil {
    f.Tag = &ast.BasicLit{}
  }
  f.Tag.Value = res
}
```

The advantage of this is that we don't run the risk of leaving an empty, non-nil tag in the tree if we encounter an error. Also, it generally is easier to follow the code if state mutation is left until the end.